src/test/java/software/amazon/smithy/lsp/ext/SmithyProjectTest.java

```diff
@@ -71,4 +75,29 @@ public void respectingEmptyConfig() throws Exception {
 
     }
 
+    @Test
+    public void definitionLocations() throws Exception {
+        Path baseDir = Paths.get(SmithyProjectTest.class.getResource("models").toURI());
+        Path modelMain = baseDir.resolve("main.smithy");
+        Path modelTest = baseDir.resolve("test.smithy");
+        List<Path> modelFiles = ImmutableList.of(modelMain, modelTest);
+
+        try (Harness hs = Harness.create(SmithyBuildExtensions.builder().build(), modelFiles)) {
+            Map<String, List<Location>> locations = hs.getProject().getLocations();
+
+            correctLocation(locations, "SingleLine", 4, 5);
```

Comment:
Seems a little strange that the starts for this are the line before, worth seeing if that's necessary to work properly.

Reply:
Positions for LSP are zero-based, so this is currently setting the end position to the start of the next line.

I'll add logic to properly determine the line length so we can determine the actual end of a shape on a given line.